test/integration/xss_protection_test.rb

```diff
@@ -0,0 +1,18 @@
+require 'test_helper'
+
+class ActionDispatch::IntegrationTest
+  include Capybara::DSL
+
+  test "should escape html correctly" do
+    visit('/bookmarks/new')
+    fill_in('Title', with: "Yamaha Saxophones")
+    fill_in('Link', with: "http://usa.yamaha.com/products/musical-instruments/winds/sax/")
+    fill_in('Description', with: "usa homepage for yamaha saxophones &`<'>\",!@$()[]{}=+")
+    click_button 'Submit'
+    bookmark = Bookmark.find_by_id("980190963")
+    assert_equal "Yamaha&nbsp;Saxophones", bookmark.title
+    assert_equal "http://usa.yamaha.com/products/musical-instruments/winds/sax/", bookmark.link
+    assert_equal "usa&nbsp;homepage&nbsp;for&nbsp;yamaha&nbsp;saxophones&nbsp;&amp;`&lt;'&gt;\&quot;,!@$()[]{}=+",
+                 bookmark.description
+  end
```

Comment:
@bion this test doesn't pass at the moment but represents my understanding of the task at hand.  I've been reading about the `safe_html` method - does rails only take care of that when using embedded ruby and now I have to do it manually because I'm using haml?  If I have to do it manually am I still able to use `safe_html` and if so, can I somehow include it in the form helper used in the `/app/views/bookmarks/_form.html.haml` partial?